packages/dds/sequence/src/test/intervalCollection.fuzzUtils.ts

```diff
@@ -138,6 +202,11 @@ export function makeReducer(loggingInfo?: LoggingInfo): Reducer<Operation, Clien
 			const collection = channel.getIntervalCollection(collectionName);
 			collection.changeProperties(id, { ...properties });
 		},
+		revertSharedStringRevertibles: async ({ channel }, { editsToRevert }) => {
+			assert(isRevertibleSharedString(channel));
+			const few = channel.revertibles.slice(0, editsToRevert);
+			revertSharedStringRevertibles(channel, few);
```

Comment:
revertibles is a stack, so you want to take from the end, not beginning. Also, the items should be removed from channel.revertibles so it can't be reverted again.

Reply:
this looks right to me now, but it brings up an interesting point that it's not obvious how `revertSharedStringRevertibles` handles its array (it calls pop() repeatedly, so last element gets reverted first). that should probably be doc'd. doesn't have to be this PR though.

Reply:
A bigger problem is that there is a missing layer that enforces the assumptions in the code and removes the need for customers to know unnecessary stuff about the implementation. There should be two separate stacks of stacks for undo and redo that are built up as edits happen, so the splicing should never happen.